content/Developer_Program/WebAuthn_Starter_Kit/Adaptive_Multi-Factor_Authentication.adoc

```diff
@@ -0,0 +1,169 @@
+ 
+As with the New Account, once the user is authenticated, they will be issued an Auth token as defined by the OpenID connect flow and be directed to the profile page.
+
+==== Account Recovery
+
+Should the user select the Account recovery option, they will be taken to a dialog allowing them to provide a Recovery code. Upon entering the Recovery Code, the user will be issued an Auth Token and be directed to the profile page. Each Recovery code may only be used once - after which, it is flagged as “used” in the database. While not implemented in the WebAuthn Starter kit, it is recommended to log as much information about the session where a Recovery Code is used as possible, in order to identify fraudulent attempts to access a protected account.
+ 
+It should be noted that the use of Recovery Codes in the WebAuthn Starter Kit was to demonstrate the possibility of adding account recovery to your   Adaptive Multi-Factor Authentication experience. For a production deployment please consider other alternatives for account recovery.
+
+==== Adding an Authenticator
+[%header,cols="1,1,1,1,1"]
+|===
+| *WebAuthn authenticator* | *Examples* | *Register a WebAuthn credential* | *Provide U2F Password* | *Set option to create resident credential*
+
+| FIDO2
+a| 
+* Yubikey 5 Series
+* YubiKey BIO
+* Face ID
+a|image::Images/group-4.png[]
+|
+|
+
+| U2F
+a| 
+* Android Biometrics and Security Key
+* YubiKey 4 Series
+a|image::Images/group-4.png[]
+a|image::Images/group-4.png[]
+|
+
+| Usernameless - FIDO2 with a Resident Credential
+a| 
+* YubiKey 5 Series
+a|image::Images/group-4.png[]
+|
+a|image::Images/group-4.png[]
+|===
+
+When adding a new authenticator to the user profile, the username associated with the account will automatically be used, without prompting the user to enter it again. When the authenticator registration begins, the WebAuthn Starter kit will first check to ensure the authenticator has not already been associated with the user account. Reused authenticators will have the registration rejected.
+ 
+As with the new account creation flow, if the authenticator supports both Discoverable Credentials and User Verification, the registration will proceed as normal down a User-Verified registration flow, with the user entering their PIN or biometric. If the authenticator does not support either Discoverable Credentials or User Verification, but a U2F Password has already been set for the user’s account, the registration will proceed, and the U2F Password will be associated with the authenticator. In the event a U2F Password has not already been provided, the user will be requested to provide one.
+ 
+Once the registration has completed, the user will be allowed to name the authenticator. Once the authenticator has been named, it will be associated with the user’s account, be able to authenticate the user during login, and be listed in the user’s profile.
```

Comment:
Is it is still true that naming the authenticator is triggered after registration ceremony is complete?